private-templates-service/adaptivecards-templating-service/src/authproviders/AzureADProvider.ts

```diff
@@ -0,0 +1,57 @@
+import { AuthenticationProvider } from "./IAuthenticationProvider";
+import jws, { Algorithm, Signature } from "jws";
+import axios from "axios";
+
+/**
+ * @class
+ * Class representing authentication with Azure AD
+ * @extends AuthenticationProvider
+ */
+export class AzureADProvider implements AuthenticationProvider {
+    private user : string = "";
+    private static CERT_URL : string = "https://login.microsoftonline.com/common/discovery/keys";
+
+    async isValid(accessToken: string) : Promise<boolean> {
```

Comment:
I imagined the AD Auth provider would make use of an existing AD node library, e.g: https://github.com/AzureAD/azure-activedirectory-library-for-nodejs

Reply:
I took a look at the AD library, but it looks like it's being used to get access tokens - since the backend library receives an access token I thought we would validate the jwt instead? Do you know if there's a way to validate an access token with AD without manually decoding the jwt?

Reply:
Just had a quick look and the example docs are doing something similar to you, ignore my comment :)